va/va.go

```diff
@@ -327,15 +372,35 @@ func (va *ValidationAuthorityImpl) validateTLSSNI01WithZName(ctx context.Context
 		},
 	}
 	if problem != nil {
-		return validationRecords, problem
+		return nil, validationRecords, problem
 	}
+	thisRecord := &validationRecords[0]
 
 	// Make a connection with SNI = nonceName
 	portString := strconv.Itoa(va.tlsPort)
 	hostPort := net.JoinHostPort(addr.String(), portString)
-	validationRecords[0].Port = portString
+	thisRecord.Port = portString
 
 	certs, problem := va.getTLSSNICerts(hostPort, identifier, challenge, zName)
+
+	// If there was a problem getting the certificates and the IPv6 first feature
+	// is enabled, then try to fetch the certificates again using a fallback
+	// address
+	if problem != nil && features.Enabled(features.IPv6First) {
```

Comment:
I'm not sure this approach is actually possible.

`tls.DialWithDialer` takes a `*net.Dialer`, which is a concrete type not an interface. the `dialer` type that is defined in the VA and returned from `resolveAndConstructDialer` isn't actually a `net.Dialer`. It has a `Dial` function that matches the `net.Dialer.Dial` signature and we do some weird trickery constructing an `http.Transport` and using it with a `http.Client` for the HTTP case. I don't see an easy way to do that with the TLS code